### PR TITLE
Feature/liayoo/staking balance total sum

### PR DESCRIPTION
**Pull request by liayoo**
- Added staking_balance_total_sum timer flag and bandage file
- Updated _stake() and _unstake() to update the balance_total_sum after the flag is enabled
- Updated getAppStakesTotalFromStateRoot() to use the balance_total_sum if available

### CODE REVIEW

File: db/functions.js [hunk 1]
```diff
@@ -973,7 +981,15 @@ class Functions {
         this.setValueOrLog(expirationPath, newExpiration, context);
       }
       const balanceTotalPath = PathUtil.getStakingBalanceTotalPath(serviceName);
+      const balanceTotalSumPath = PathUtil.getStakingBalanceTotalSumPath();
       this.incValueOrLog(balanceTotalPath, value, context);
+      if (isEnabledTimerFlag('staking_balance_total_sum', context.blockNumber)) {
+        if (this.db.getValue(balanceTotalSumPath) === null) {
```
Comment (platfowner) [inline db/functions.js:987]: 👍

File: db/index.js [hunk 2]
```diff
@@ -818,6 +818,13 @@ class DB {
   }
 
   static getAppStakesTotalFromStateRoot(stateRoot) {
+    const balanceTotalSum = DB.getValueFromStateRoot(
+        stateRoot, PathUtil.getStakingBalanceTotalSumPath());
+    const consensusStakeTotal = DB.getValueFromStateRoot(
+        stateRoot, PathUtil.getStakingBalanceTotalPath(PredefinedDbPaths.CONSENSUS)) || 0;
+    if (balanceTotalSum !== null) {
```
Comment (platfowner) [inline db/index.js:825]: minor) could add some comments for 'staking_balance_total_sum' timer flag here?

File: test/unit/consensus.test.js [hunk 3]
```diff
@@ -19,6 +19,8 @@ describe("Consensus", () => {
     await setNodeForTesting(node1, 0);
     node2 = new BlockchainNode();
     await setNodeForTesting(node2, 1);
+    // to advance block number to 2 and apply the bandages
```
Comment (platfowner) [inline test/unit/consensus.test.js:22]: 👍

File: test/unit/consensus.test.js [hunk 4]
```diff
@@ -142,7 +144,7 @@ describe("Consensus", () => {
     const proposeWithStakeLessThanMin = getTransaction(node2, {
         operation: {
           type: 'SET_VALUE',
-          ref: `/consensus/number/${lastBlock.number + 1}/propose/${addr}`,
```
Comment (platfowner) [inline test/unit/consensus.test.js:145]: 👀

**Review by platfowner — APPROVED** [review]
LGTM with some minor comments.
Thanks for the nice implementation.

File: test/unit/consensus.test.js (hunk 4 above)
Comment (liayoo) [reply test/unit/consensus.test.js:145]: The paths were incorrect in the unit tests but didn't get caught until the recently introduced path length check..!

File: db/index.js (hunk 2 above)
Comment (liayoo) [reply db/index.js:825]: done!

**Comment by liayoo** [comment]
Thanks for the review!! Now merging~

**Pull request merged**